### ISSUE TITLE
List components: Missing screen reader index count

### ISSUE THREAD

**Issue by khmakoto** [microsoft/fluentui#15355]
There are a couple of issues regarding the index count of items in `List` components:
- `aria-posinset` and `aria-setsize` attributes are being flagged as not allowed in group headers given that they have the `row` role to account for correct `grid` semantics. We should remove them and use `aria-rowindex` and `aria-rowcount` instead.
- Index count is missing for items within a group so the total number of them are not read when navigating to the group (see #11817) nor are their positions in the set read when navigating through them (see #11819). This can be fixed by passing the correct `row` role to these items and setting both `aria-rowindex` and `aria-rowcount`.
- We've also been talking about having the items in the list match the visual display rather than including hidden/collapsed rows and having the same index across both headers and items. This is a tentative solution that is pending further discussion before committing to it. This can be evaluated by the following example where `Group Header 2` is collapsed:

```
Group Header 1      -   rowindex=1   rowcount=6
    List item 1.1   -   rowindex=2   rowcount=6
    List item 1.2   -   rowindex=3   rowcount=6
Group Header 2      -   rowindex=4   rowcount=6
Group Header 3      -   rowindex=5   rowcount=6
    List item 3.1   -   rowindex=6   rowcount=6
```

And when we expand it then the attributes change to reflect it:

```
Group Header 1      -   rowindex=1   rowcount=8
    List item 1.1   -   rowindex=2   rowcount=8
    List item 1.2   -   rowindex=3   rowcount=8
Group Header 2      -   rowindex=4   rowcount=8
    List item 2.1   -   rowindex=5   rowcount=8
    List item 2.2   -   rowindex=6   rowcount=8
Group Header 3      -   rowindex=7   rowcount=8
    List item 3.1   -   rowindex=8   rowcount=8
```

**Comment by JustSlone** [comment]
@khmakoto, does this issue cover the Accessibility Insights issue found on: https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist/grouped



Path:  
- div[aria-posinset="\32 "]

Snippet: 
- `<div class="ms-GroupHeader root-318" role="row" aria-setsize="3" aria-posinset="2" data-is-focusable="true" aria-label="Color: &quot;green&quot;" aria-expanded="true" aria-level="1">`

How to Fix: 
- Fix the following:ARIA attributes are not allowed: aria-setsize="3", aria-posinset="2"

**Comment by khmakoto** [comment]
@JustSlone yes, this issue definitely covers those, specifically in the first bullet point of its description.

**Comment by bcoard** [comment]
Hey folks, do we have any plan/movement here? I've experienced 2 different behaviors:

1. On my local deployment I can hear n of m information being read out but it reads the indexs as though it's a flat list when ideally the first list element for each child should reset the index to 1 (e.g. first child in a group should be "1 of m").
2. On the grouped list in the example page, I don't hear n of m at all.

Is this actually an issue with Edge or this control?
